blueocean-core-js/src/js/fetch.ts

```diff
@@ -59,26 +76,24 @@ export const FetchFunctions = {
      * This method checks for for 2XX http codes. Throws error it it is not.
      * This should only be used if not using fetch or fetchJson.
      */
-    checkStatus(response) {
+    checkStatus(response: Response): Response {
         if (response.status >= 300 || response.status < 200) {
             const message = `fetch failed: ${response.status} for ${response.url}`;
-            const error = new Error(message);
-            error.response = response;
-            throw error;
+            throw {message , response};
```

Comment:
We should be throwing `Error` instances, not objects. The stack trace comes from the object creation, not the throw statement.